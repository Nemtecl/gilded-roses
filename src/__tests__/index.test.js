const { Item, Shop } = require("../");

const ITEM_NAMES = [
  "Aged Brie",
  "Backstage passes to a TAFKAL80ETC concert",
  "Sulfuras, Hand of Ragnaros",
];

const ITEM_SELLINS = [-1, 0, 10, 20, 30, 40, 50];
const QUALITIES = [-1, 0, 49, 80];

const ITEMS = ITEM_NAMES.map((name) =>
  ITEM_SELLINS.map((sellIn) =>
    QUALITIES.map((quality) => new Item(name, sellIn, quality))
  )
).flat(3);

test("gilded roses", () => {
  const shop = new Shop(ITEMS);

  const updatedItems = shop.updateQuality();

  expect(updatedItems).toMatchSnapshot();
});
