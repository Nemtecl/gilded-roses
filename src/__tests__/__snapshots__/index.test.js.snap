// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gilded roses 1`] = `
[
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": -2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 2,
    "sellIn": -2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": -2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 80,
    "sellIn": -2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": -1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 2,
    "sellIn": -1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": -1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 80,
    "sellIn": 9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 19,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 19,
  },
  Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 19,
  },
  Item {
    "name": "Aged Brie",
    "quality": 80,
    "sellIn": 19,
  },
  Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 29,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 29,
  },
  Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 29,
  },
  Item {
    "name": "Aged Brie",
    "quality": 80,
    "sellIn": 29,
  },
  Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 39,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 39,
  },
  Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 39,
  },
  Item {
    "name": "Aged Brie",
    "quality": 80,
    "sellIn": 39,
  },
  Item {
    "name": "Aged Brie",
    "quality": 0,
    "sellIn": 49,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 49,
  },
  Item {
    "name": "Aged Brie",
    "quality": 50,
    "sellIn": 49,
  },
  Item {
    "name": "Aged Brie",
    "quality": 80,
    "sellIn": 49,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 1,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 2,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 80,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": 19,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 1,
    "sellIn": 19,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 19,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 80,
    "sellIn": 19,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": 29,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 1,
    "sellIn": 29,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 29,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 80,
    "sellIn": 29,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": 39,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 1,
    "sellIn": 39,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 39,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 80,
    "sellIn": 39,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": 49,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 1,
    "sellIn": 49,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 49,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 80,
    "sellIn": 49,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": -1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": -1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 10,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 10,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 10,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 10,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 20,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 20,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 20,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 20,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 30,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 30,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 30,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 30,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 40,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 40,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 40,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 40,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": -1,
    "sellIn": 50,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 0,
    "sellIn": 50,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 49,
    "sellIn": 50,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 50,
  },
]
`;
